packages/api/src/Api.ts

```diff
@@ -31,9 +31,26 @@ import logger from './util/logging';
 
 const Types = require('@cennznet/types');
 
+export const DEFAULT_TIMEOUT = 30000;
+
 export class Api extends ApiPromise {
-    static async create(options: ApiOptions | ProviderInterface = {}): Promise<Api> {
-        return new Api(options).isReady;
+    static async create(provider: ApiOptions | ProviderInterface = {}): Promise<Api> {
+        const options =
```

Comment:
you should only try to extract timeout if options is ApiOptions, don't convert it here.